pulsar4s-core/src/main/scala/com/sksamuel/pulsar4s/ProducerConfig.scala

```diff
@@ -23,7 +23,8 @@ case class ProducerConfig(topic: Topic,
                           roundRobinRouterBatchingPartitionSwitchFrequency: Option[Int] = None,
                           producerName: Option[String] = None,
                           sendTimeout: Option[FiniteDuration] = None,
-                          compressionType: Option[CompressionType] = None)
+                          compressionType: Option[CompressionType] = None,
+                          additionalProperties: Option[Map[String, AnyRef]] = None)
```

Comment:
This could perhaps be `Map[String, String]` and default to empty.

Reply:
I was not sure about defaulting to empty, I fix this and push